libraries/Microsoft.Bot.Expressions.Properties/Extensions.cs

```diff
@@ -35,12 +36,13 @@ public static object Evaluate(this LGFile lgFile, string inlineStr, object scope
             inlineStr = !inlineStr.Trim().StartsWith(multiLineMark) && inlineStr.Contains('\n')
                    ? $"{multiLineMark}{inlineStr}{multiLineMark}" : inlineStr;
 
-            var wrappedStr = $"# {fakeTemplateId} \r\n - {inlineStr}";
-
-            var newContent = $"{lgFile.Content}\r\n{wrappedStr}";
+            var newContent = $"# {fakeTemplateId} \r\n - {inlineStr}";
 
             var newLgFile = LGParser.ParseText(newContent, lgFile.Id, lgFile.ImportResolver);
-            return newLgFile.EvaluateTemplate(fakeTemplateId, scope);
+
+            var allTemplates = lgFile.AllTemplates.Union(newLgFile.AllTemplates).ToList();
```

Comment:
couldn't you precompute this and only redo when new templates are added?

Reply:
this newLgFile is computed from the parameter `inlineStr`, so we can't pre-compute it.  What we can do is to cache this string, to avoid re-compute, but based on the profiling result, not much improvement here, because the inlineStr is very small. So this is good enough for now.